src/Features/Core/Portable/NameTupleElement/AbstractNameTupleElementCodeRefactoringProvider.cs

```diff
@@ -43,29 +44,11 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
                 return default;
             }
 
-            if (span.Length > 0)
-            {
-                return default;
-            }
-
-            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-            var position = span.Start;
-            var token = root.FindToken(position);
-            if (token.Span.Start == position &&
-                IsCloseParenOrComma(token))
-            {
-                token = token.GetPreviousToken();
-                if (token.Span.End != position)
-                {
-                    return default;
-                }
-            }
-
+            var helperService = document.GetLanguageService<IRefactoringHelpersService>();
             var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
-            var argument = root.FindNode(token.Span)
-                .GetAncestorsOrThis<TArgumentSyntax>()
-                .FirstOrDefault(node => syntaxFacts.IsTupleExpression(node.Parent));
 
+            Func<SyntaxNode, bool> isTupleArgumentExpression = n => n.Parent != null && syntaxFacts.IsTupleExpression(n.Parent);
+            var argument = await document.TryGetSelectedNodeAsync<TArgumentSyntax>(span, isTupleArgumentExpression, cancellationToken).ConfigureAwait(false);
```

Comment:
fwiw, i think this would actually read better as:

```c#
var arguments = await document.TryGetSelectedNodeAsync<TArgumentSyntax>(span, cancellationToken).ConfigureAwait(false);
var argument = arguments.FirstOrDefault(n => syntaxFacts.IsTupleExpression(n.Parent));
```

Reply:
oh, based on my comment far above, can this be:

```c#
var exprs= await document.TryGetSelectedNodeAsync<TExpressionSyntax>(span, cancellationToken).ConfigureAwait(false);
var expr = arguments.FirstOrDefault(n => n.Parent is TArgumentSyntax && n.Parent.Parent is TTupleExpression);
var argument = expr?.Parent as TArgumentSyntax;
```

?